designcompose/src/main/java/com/android/designcompose/Utils.kt

```diff
@@ -590,6 +590,122 @@ internal fun mergeStyles(base: ViewStyle, override: ViewStyle): ViewStyle {
+    // for layout.
+    builder.display_type = display_type
+    builder.position_type = position_type
+    builder.flex_direction = flex_direction
+    builder.flex_wrap = flex_wrap
+    builder.grid_layout = grid_layout
+    builder.grid_columns_rows = grid_columns_rows
+    builder.grid_adaptive_min_size = grid_adaptive_min_size
+    builder.grid_span_content = grid_span_content
+    builder.overflow = overflow
+    builder.max_children = max_children
+    builder.overflow_node_id = overflow_node_id
+    builder.overflow_node_name = overflow_node_name
+    builder.align_items = align_items
+    builder.align_self = align_self
+    builder.align_content = align_content
+    builder.justify_content = justify_content
+    builder.top = top
+    builder.left = left
+    builder.bottom = bottom
+    builder.right = right
+    builder.margin = margin
+    builder.padding = padding
+    builder.item_spacing = item_spacing
+    builder.cross_axis_item_spacing = cross_axis_item_spacing
+    builder.flex_grow = flex_grow
+    builder.flex_shrink = flex_shrink
+    builder.flex_basis = flex_basis
+    builder.bounding_box = bounding_box
+    builder.horizontal_sizing = horizontal_sizing
+    builder.vertical_sizing = vertical_sizing
+    builder.width = width
+    builder.height = height
+    builder.min_width = min_width
+    builder.min_height = min_height
+    builder.max_width = max_width
+    builder.max_height = max_height
+    builder.aspect_ratio = aspect_ratio
+    builder.pointer_events = pointer_events
+    builder.meter_data = meter_data
+    return builder
+}
+
+// Take the external layout fields of ViewStyle and put them into an ExternalLayoutData object.
+internal fun ViewStyle.externalLayoutData(): ExternalLayoutData {
+    return ExternalLayoutData(
+        margin,
+        top,
+        left,
+        bottom,
+        right,
+        // If width/height are fixed point values, don't override them since we want to use the
+        // new node's width/height. If not fixed point values, use the values in order to support
+        // auto layout properties such as constraints.
+        if (width !is Dimension.Points) width else null,
+        if (height !is Dimension.Points) height else null,
+        flex_grow,
+        flex_basis,
+        align_self,
+        position_type,
+        transform,
+        relative_transform,
+    )
+}
+
+// Take the external layout data and merge it into this ViewStyle, overriding its values and
+// returning a new ViewStyle.
+internal fun ViewStyle.withExternalLayoutData(data: ExternalLayoutData): ViewStyle {
+    val overrideStyle = this.asBuilder()
+    overrideStyle.margin = data.margin
+    overrideStyle.top = data.top
+    overrideStyle.left = data.left
+    overrideStyle.bottom = data.bottom
+    overrideStyle.right = data.right
+    // Only override the width and height if not null
+    data.width?.let { overrideStyle.width = it }
```

Comment:
heh, I think this is idiomatic Kotlin, but it took me a few tries to understand it...

Reply:
Yea there's also the `a?.let { b } ?: c`